improver/regrid/landsea.py

```diff
@@ -257,42 +264,25 @@ def __init__(
         self.regridder = Nearest(extrapolation_mode=extrapolation_mode)
         self.vicinity = OccurrenceWithinVicinity(vicinity_radius)
 
-    def correct_where_input_true(self, selector_val: int) -> None:
-        """
-        Replace points in the output_cube where output_land matches the
-        selector_val and the input_land does not match, but has matching
-        points in the vicinity, with the nearest matching point in the
-        vicinity in the original nearest_cube.
-        Updates self.output_cube.data.
-
-        Args:
-            selector_val:
-                Value of mask to replace if needed.
-                Intended to be 1 for filling land points near the coast
-                and 0 for filling sea points near the coast.
-        """
+    @functools.lru_cache(maxsize=2)
+    def _get_matches(
+        self, selector_val: int
+    ) -> Tuple[ndarray, ndarray, ndarray, ndarray]:
         # Find all points on output grid matching selector_val
         use_points = np.where(self.input_land.data == selector_val)
+        no_use_points = np.where(self.input_land.data != selector_val)
 
         # If there are no matching points on the input grid, no alteration can
-        # be made. This tests the size of the y-coordinate of use_points.
+        # be made.
         if use_points[0].size == 0:
-            return
-
-        # Get shape of output grid
-        ynum, xnum = self.output_land.shape
+            raise self._NoMatchesError
```

Comment:
This ensures that the condition of not having a match is explicit rather than implicit (implicit via a None return or similar).  It also ensures that `_get_matches` always returns a tuple of 3 cubes making the interface easier to understand.
I'm not wedded to the approach if you prefer an alternative?

Reply:
They seem like good reasons and I'm happy. The motivation for the question was to understand whether the traceback that one gets when performing this in the terminal going to clog the logs, or is it lost when running in a suite context? That was my primary concern, but I assume we are not raising this exception often so perhaps its a moot point.

Reply:
> ...whether the traceback that one gets when performing this in the terminal is going to clog the logs...

You get the traceback running this on the terminal?
Since the exception is caught by the public method, I wasn't expecting anything to appear in any logs/stdout...